android/src/main/java/com/RNTextInputMask/RNTextInputMaskModule.kt

```diff
@@ -39,8 +53,22 @@ class RNTextInputMaskModule(private val context: ReactApplicationContext) : Reac
     fun unmask(maskString: String?,
                inputValue: String,
                autocomplete: Boolean,
+               options: ReadableMap,
                promise: Promise) {
-        val mask = Mask(maskString!!)
+             val customNotations = options.list("customNotations") { array, index ->
+            val raw = array.getMap(index) ?: throw IllegalArgumentException("could not parse notation")
+            Notation(
+                character = raw.string("character")?.first()
+                    ?: throw IllegalArgumentException("character is required for notation"),
+                characterSet = raw.string("characterSet")
+                    ?: throw IllegalArgumentException("characterSet is required for notation"),
+                isOptional = raw.boolean("isOptional")
+                    ?: throw IllegalArgumentException("isOptional is required for notation")
+            )
+        }
+        val safeMaskString = maskString?.let { it } ?: return
+        val notations = customNotations ?: listOf<Notation>()
+        val mask = Mask(safeMaskString, notations)
```

Comment:
I'd pull this into a utility function or extension instead for reusability. Something like:

```kotlin
fun createMask(mask: String?, options: ReadableMap): Mask? {
 // code goes here
}
```
Then you could do something like:
```kotlin
val mask = createMask(maskString, options) ?: return
```
And reuse this everywhere this logic has been duplicated